### ISSUE TITLE
Add option to include child processes in process collector

### ISSUE THREAD

**Issue by SuperQ** [prometheus/client_golang#759]
The NewProcessCollector currently only looks at one PID.

It would be useful to have a boolean flag that would allow aggregating some of the metrics. This would be useful for creating exporters for things that spawn un-privileged workers from a main PID.

For example, haproxy, postfix, apache, etc.

To be debated, how to handle the labels/aggregations here.

**Comment by beorn7** [comment]
Yeah, aggregations will be problematic for some metrics.

How about just calling `NewProcessCollector` n-times and register it via a label-wrapped registry using  https://pkg.go.dev/github.com/prometheus/client_golang/prometheus@v1.6.0?tab=doc#WrapRegistererWith ?

**Comment by SuperQ** [comment]
Yes, that may be an option. I'd have to think about how to deal with changing child PIDs.
